ENH: Improve quotes display CSS

12rambau opened the pull request:
Fix #454

The quotes were using a legacy color. 
Used the now-classic trick of admonition to display quotes in light grey using a left border (grey as well. It looks more like what we see on other platforms like GitHub and is working with any quote directive including `epigraph`, `highlight` and `pull-quote`.
<img width="1001" alt="Capture d’écran 2022-06-20 à 10 21 20" src="https://user-images.githubusercontent.com/12596392/174558504-3c861640-f1c0-49dd-9ad2-ae701bc52fc2.png">

choldgraf commented on src/pydata_sphinx_theme/assets/styles/content/_quotes.scss:
```diff
@@ -2,5 +2,25 @@
+
+  // This is a hack to control the opacity for admonitions with our color variables
+  // ref: https://stackoverflow.com/a/56951626/6734243
+  &:after {
```
Just a note now that we've got this used in many places in the theme, I think that we could turn this into a `@mixin` if we wanted. Then we could just use `include` to add this behavior. E.g., something like
```scss
@mixin background-from-color-variable($color-variable) {
  // This is a hack to control the opacity for admonitions with our color variables
  // ref: https://stackoverflow.com/a/56951626/6734243
  &:after {
    content: "";
    width: 100%;
    height: 100%;
    position: absolute;
    left: 0;
    top: 0;
    background-color: var(#{$color-variable});
    background-color: var(--pst-color-border);
    opacity: 0.1;
    border-radius: 0.2rem;
  }
}
```

and then call it with something like:

```scss
blockquote {
  ...css rules
  @include background-from-color-variable(--pst-color-border);
}
```

for ref: https://sass-lang.com/documentation/at-rules/mixin

choldgraf reviewed (commented):
This looks good in general - two quick thoughts:

1. The text doesn't seem to be selectable, I wonder if the `&after` is floating above it in the blockquotes?
2. Could we add some padding just above and below the text? I think we could just re-use the same top/bottom padding that github uses

> hi 
> there
> - Steve

choldgraf commented:
Looking better! Three quick thoughts:

1. The padding in literal blocks is twice as much as others.

   <img width="352" alt="chrome_R2kEFtDmOJ" src="https://user-images.githubusercontent.com/1839645/174747272-0b059e8e-4cff-40bb-83a7-effe154c0d0b.png">

1. There seems to be more padding on the bottom than the top. I think we should remove the bottom margin only for the last element in the quote. For example:
   
   <img width="343" alt="chrome_NVYMyPTPVJ" src="https://user-images.githubusercontent.com/1839645/174747147-c54d149a-713c-4d2c-8709-7452f792bc5c.png">

1. Also just a note I still cannot select text. When I hover over the quote block, my cursor does not register as hovering over text. I think something exists *above* the text in the quote blocks

12rambau commented:
> The padding in literal blocks is twice as much as others

I removed to margins of the `line-block` when they are in `<blockquote>`.

> There seems to be more padding on the bottom than the top

There is a margin-bottom associated with `<p>` element in the theme. It's now removed for the last child

> Also just a note I still cannot select text.

There is nothing on top of the `<blockquote>`. They are behaving the same way as the title of admonitions and deprecated blocks i.e. the cursor stay unchanged but the text can be copied as usual (at least on my computer using the latest Safari browser). I assume it's related to the `::before` pseudo element that forces the cursor.

choldgraf commented on src/pydata_sphinx_theme/assets/styles/abstracts/_mixins.scss:
```diff
@@ -21,12 +21,32 @@
+    height: 100%;
+    position: absolute;
+    left: 0;
+    top: 0;
```
I think that this will make the text highlightable more easily, will merge in and see if that works

choldgraf reviewed (commented):
Just one quick comment in there, if that resolves it then I think we should merge

choldgraf commented:
OK so setting `z-index: -1` does fix the text selection problem, but it results in the admonitions no-longer having colored title areas. That is because the admonitions *also* set a background color, and this is then "higher" than the title area and so covers that color entirely.

I think there are 3 ways we could get around this:

1. Define the `z-index: -1` as a special-case hack just for the quote blocks (or other blocks where we want the text selectable).
2. Remove `background-color` from the admonitions and just have them be transparent 
3. Find a way to make the `z-index` of the admonition background still be behind the admonition title

I think it's going to confuse people that they can't select text inside of quotes, so IMO it's something we should try to fix

12rambau commented:
We initially set back the admonition background when we were playing with the colors to increase the contrast in the dark background so I would say we really want to keep it. I will go with the simple z-index hack for this specific use case. it's harmless and works just fine.

choldgraf commented on src/pydata_sphinx_theme/assets/styles/abstracts/_mixins.scss:
```diff
@@ -21,12 +21,33 @@
+    position: absolute;
+    left: 0;
+    top: 0;
+    z-index: -1;
```
```suggestion
```

choldgraf commented on src/pydata_sphinx_theme/assets/styles/abstracts/_mixins.scss:
```diff
@@ -21,12 +21,32 @@
 /**
-  * Hide the scrollbar until the element is overed, so keep the page clean
-  */
+* Hide the scrollbar until the element is overed, so keep the page clean
+*/
```
```suggestion
 * Hide the scrollbar until the element is overed, so keep the page clean
 */
```

choldgraf commented on src/pydata_sphinx_theme/assets/styles/abstracts/_mixins.scss:
```diff
@@ -21,12 +21,32 @@
   }
 }
+
+/**
+* create a low opacity background behind object using our variable colors
+*/
```
```suggestion
/**
 * create a low opacity background behind object using our variable colors
 */
```
